salt/modules/virt.py

```diff
@@ -909,7 +855,7 @@ def _disk_profile(profile, hypervisor, disks=None, vm_name=None, image=None, poo
         overlay = {'format': 'vmdk',
                    'model': 'scsi',
                    'device': 'disk',
-                   'pool': '[{0}] '.format(pool if pool else '0')}
+                   'pool': '[0] '}
```

Comment:
Shouldn't we remove the `pool` key from here and also from `libvirt_volume.jinja` template?

Reply:
Yes and no. Yes we should remove it from here because this is ESX-specific code and should be handled somewhere else. No, because there is still a pool key on disks structures that replace the `virt.init()` `pool` argument.

Anyway I am currently working on another PR that will completely rework how volumes are handled in the virt module and will clean that mess as well.